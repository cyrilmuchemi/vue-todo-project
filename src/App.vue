<template>
  <TopBar />
  <TodoInput />
  <TodoList />
  <p class="footer">KaziQuest Interview by Cyril Muchemi.</p>
</template>

<script setup>
import TopBar from "./components/TopBar.vue";
import TodoInput from "./components/TodoInput.vue";
import TodoList from "./components/TodoList.vue";
</script>
