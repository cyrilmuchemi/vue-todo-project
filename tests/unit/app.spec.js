import { mount } from "@vue/test-utils";
import App from "./../../src/App.vue";

describe("Mounted App", () => {

  it('renders correctly', () => {
  const wrapper = mount(App, {
    addTodo(title) {
      let todo = { title: title, completed: false };
      todos.value.push(todo);
      localStorage.setItem("todos", JSON.stringify(todos.value));
    }
  })
  expect(wrapper.element).toMatchSnapshot()
  })
})








