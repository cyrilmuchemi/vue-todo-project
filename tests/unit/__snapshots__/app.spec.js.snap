// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted App renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="topbar | row"
  >
    <h1>
      TODO
    </h1>
    <img
      alt="Toggle Theme"
      src="../src/assets/images/ICON-MOON.SVG"
    />
  </div>
  <div
    class="todo-input | todo-item card"
  >
    <div
      class="checkbox"
    />
    <input
      class="input"
      placeholder="Create a new todo.."
      type="text"
    />
  </div>
  
  <div
    class="todo-list | card"
  >
    
    
    <div
      class="menu | row"
    >
      <h6>
        0 items left
      </h6>
      <div
        class="inner-row bold"
      >
        
        <p
          class="menu-active"
        >
          All
        </p>
        <p
          class=""
        >
          Active
        </p>
        <p
          class=""
        >
          Completed
        </p>
        
      </div>
      <p>
        Clear Completed
      </p>
    </div>
  </div>
  <div
    class="detached-menu | card row"
  >
    
    <p
      class="menu-active"
    >
      All
    </p>
    <p
      class=""
    >
      Active
    </p>
    <p
      class=""
    >
      Completed
    </p>
    
  </div>
  
  <p
    class="footer"
  >
    KaziQuest Interview by Cyril Muchemi.
  </p>
  
</div>
`;
